src/git_util.rs

```diff
@@ -27,6 +27,11 @@ fn make_credentials_callback(
     let mut token_tried = false;
 
     move |url, username, allowed_types| {
+        if allowed_types.contains(CredentialType::SSH_KEY) {
+            let username = username.unwrap();
```

Comment:
We probably shouldn't unwrap here, but rather return some sort of error.

Reply:
Cargo's implementation doesn't, having a comment saying it doesn't get called if there is no username as the reasoning. So, I just followed Cargo's implementation since that's what all of Wally's was based on, to begin with.
https://github.com/rust-lang/cargo/blob/79b397d72c557eb6444a2ba0dc00a211a226a35a/src/cargo/sources/git/utils.rs#L637-L646